language/move-model/bytecode/src/stackless_bytecode_generator.rs

```diff
@@ -327,17 +327,15 @@ impl<'a> StacklessBytecodeGenerator<'a> {
             },
 
             MoveBytecode::Abort => {
-                if self.func_env.is_entry() {
-                    let acquires = self.func_env.get_acquires_global_resources();
-                    if let Some(acquires) = acquires {
-                        for acquire in acquires {
-                            debug!("acquire global resource: {acquire:?}");
-                            let indices = borrow_map.get(&acquire).unwrap_or(&vec![]).clone();
-                            debug!("acquire indices: {indices:?}");
-                            if !indices.is_empty() {
-                                debug!("emitting drop for acquire: {acquire:?}");
-                                self.code.push(mk_call(Operation::Release, vec![], indices));
-                            }
+                let acquires = self.func_env.get_acquires_global_resources();
+                if let Some(acquires) = acquires {
+                    for acquire in acquires {
+                        debug!("acquire global resource: {acquire:?}");
+                        let indices = borrow_map.get(&acquire).unwrap_or(&vec![]).clone();
```

Comment:
Just an idea to avoid using clone on empty vec - would something like that work?

```suggestion
                        let indices = borrow_map.get(&acquire).cloned().unwrap_or_default();
```

Reply:
unfortunately this does not work here